app/src/main/java/com/ataulm/artcollector/gallery/ui/GalleryAdapter.kt

```diff
@@ -3,23 +3,26 @@ package com.ataulm.artcollector.gallery.ui
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.paging.PagingDataAdapter
 import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
 import com.ataulm.artcollector.R
 import com.bumptech.glide.RequestManager
 import kotlinx.android.synthetic.main.itemview_painting.view.*
 
-internal class GalleryAdapter constructor(
-        private val glideRequestManager: RequestManager
-) : ListAdapter<UiPainting, GalleryAdapter.PaintingViewHolder>(PaintingDiffer) {
+internal class GalleryAdapter(private val glideRequestManager: RequestManager)
+    : PagingDataAdapter<UiPainting, GalleryAdapter.PaintingViewHolder>(PaintingDiffer) {
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PaintingViewHolder {
         val view = LayoutInflater.from(parent.context).inflate(R.layout.itemview_painting, parent, false)
         return PaintingViewHolder(glideRequestManager, view)
     }
 
-    override fun onBindViewHolder(viewHolder: PaintingViewHolder, position: Int) = viewHolder.bind(getItem(position))
+    override fun onBindViewHolder(viewHolder: PaintingViewHolder, position: Int) {
+        // the item might be null while loading - TODO: placeholders
+        getItem(position)?.let { viewHolder.bind(it) }
```

Comment:
biggest difference is that `getItem` might be null depending on whether or not placeholders were enabled. watch this space for placeholders shiz.